stellargraph/data/explorer.py

```diff
@@ -953,3 +954,31 @@ def _walk(self, src, dst, t, length, bias_type, np_rs):
                 break
 
         return walk
+
+
+class ConcurrentGraphWalk:
```

Comment:
Instead of `GraphWalk`, maybe this could be:

```suggestion
class ConcurrentSeeded:
```

or

```suggestion
class SeededPerBatch:
```

because it'll be useful in `KGTripleSequence` in https://github.com/stellargraph/stellargraph/pull/848 too. (The docs will need updating too.)

It could potentially move to the `random` module too?